lib/IDE/IDETypeChecking.cpp

```diff
@@ -808,6 +808,20 @@ class VariableTypeCollector : public SourceEntityWalker {
       auto TyOffset = getTypeOffset(Buffer.str());
       bool HasExplicitType =
           VD->getTypeReprOrParentPatternTypeRepr() != nullptr;
+      // Handle typed if/guard/while-let as a special case (e.g. `if let x: Int
+      // = Optional.some(4)`), since the `TypedPattern` is not the top-level
+      // pattern here - instead it is an implicit `OptionalSomePattern`
+      if (!HasExplicitType) {
+        if (auto *somePattern =
+                dyn_cast_or_null<OptionalSomePattern>(VD->getParentPattern())) {
+          if (somePattern->isImplicit()) {
```

Comment:
Is it strictly necessary to check if the `OptionalSomePattern` is implicit here?

Reply:
I don't think that having a typed pattern in an explicit pattern is possible in Swift currently, I included this check just to be sure that we consider the expected case only though.